### PR TITLE
Risk calc unit tests and bug fixes (EXPOSUREAPP-5697)

### PR DESCRIPTION
**Pull request by chiljamgossow**
Bug fixes and unit tests

- Moved logic of reporting calculation success and failure from matcher to repository
- fixes bug in combining risk results
- adds db clearing on data reset

Still missing: unit test for PresenceTracingRiskRepository

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/PresenceTracingRiskRepository.kt [hunk 1]
```diff
@@ -41,84 +42,124 @@ class PresenceTracingRiskRepository @Inject constructor(
         database.presenceTracingRiskLevelResultDao()
     }
 
-    private val allMatches = traceTimeIntervalMatchDao.allMatches().map { list ->
-        list.map {
-            it.toModel()
+    private val matchesOfLast14DaysPlusToday = traceTimeIntervalMatchDao.allMatches()
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/PresenceTracingRiskRepository.kt:45]: Makes sure only matches of the last 14 days plus today are considered

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/RiskLevelResultExtensions.kt [hunk 2]
```diff
@@ -24,40 +22,6 @@ data class DisplayableEwRiskResults(
     val lastSuccessfullyCalculated: EwRiskLevelResult
 )
 
-fun List<CombinedEwPtRiskLevelResult>.tryLatestResultsWithDefaults(): DisplayableRiskResults {
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/RiskLevelResultExtensions.kt:27]: Not needed anymore. Logic is now covered in BaseRiskLevelStorage

**Review by d4rken — APPROVED** [review]
lgtm, the tests are awesome !

**Comment by d4rken** [comment]
To make it easier for me to resolve conflicts and overlap in refactoring, this PR was merged into #2707.

**Pull request closed without merging**